packages/client/src/Recovery.ts

```diff
@@ -339,13 +381,155 @@ export class PendingProtection implements WithProtectionParameters {
+    }
+
+    readonly id: string;
+    readonly requesterAddress: string;
+    readonly decision: LegalOfficerDecision;
+    readonly userIdentity: UserIdentity;
+    readonly userPostalAddress: PostalAddress;
+    readonly createdOn: string;
+    readonly isRecovery: boolean;
+    readonly addressToRecover: string | null;
+    readonly status: ProtectionRequestStatus;
+    readonly legalOfficerAddress: string;
+    readonly otherLegalOfficerAddress: string;
+
+    protected readonly loRecoveryClient: LoRecoveryClient;
+}
+
+export class CancellableRequest extends ProtectionRequestImpl {
+
+    cancel(): Promise<void> {
+        return this.loRecoveryClient.cancel({
+            id: this.id
+        })
+    }
+}
+
+export class UpdatableRequest extends CancellableRequest {
+
+    update(parameters: UpdateParameters): Promise<void> {
+        return this.loRecoveryClient.update({
+            id: this.id,
+            ...parameters
+        })
+    }
+}
+
+export class ReSubmittableRequest extends CancellableRequest {
+
+    resubmit(): Promise<void> {
+        return this.loRecoveryClient.resubmit({
+            id: this.id
+        })
+    }
+}
+
+export class UpdatableReSubmittableRequest extends UpdatableRequest {
+
+    resubmit(): Promise<void> {
+        return this.loRecoveryClient.resubmit({
+            id: this.id
+        })
+    }
+}
+
+export class RejectedRecovery implements WithProtectionParameters {
+
+    constructor(sharedState: RecoverySharedState) {
+        this.sharedState = sharedState;
+    }
+
+    protected readonly sharedState: RecoverySharedState;
+
+    get protectionParameters(): ProtectionParameters {
+        return buildProtectionParameters(this.sharedState);
+    }
+
+    async cancel(): Promise<NoProtection> {
+        return Promise.all(this.sharedState.allRequests.map(request => {
+            if (request instanceof CancellableRequest) {
+                request.cancel()
+            } else {
+                return Promise.reject("Request cannot be cancelled.")
+            }
+        })).then(() => {
+            return new NoProtection(this.sharedState).refresh();
+        })
+    }
+
+    async resubmit(currentLegalOfficer: LegalOfficer): Promise<PendingProtection> {
+        const request = this.sharedState.allRequests.find(request => request.legalOfficerAddress === currentLegalOfficer.address);
+        if (request && (request instanceof ReSubmittableRequest || request instanceof UpdatableReSubmittableRequest)) {
+            await request.resubmit();
+            return await new PendingProtection(this.sharedState).refresh() as PendingProtection
+        }
+        throw new Error("Unable to find the request to resubmit")
+    }
+
+}
+
+
+export class RejectedProtection extends RejectedRecovery {
+
+    async changeLegalOfficer(currentLegalOfficer: LegalOfficer, newLegalOfficer: LegalOfficer): Promise<PendingProtection> {
+
+        // 1. Cancel the current request
```

Comment:
For each comment of this kind, write a private method and name it according to the comment.

Reply:
Good point, code adapted.